### ISSUE TITLE
Odometry (Python API) gives erroneous warning and doesn't compute odometry

### ISSUE THREAD

**Issue by scimad** [isl-org/Open3D#1356]
I am trying to compute odometry from two RGBD images. So, I tried to read two consecutive RGBD frames from Azure Kinect and use the `open3d.odometry.compute_rgbd_odometry()` function to get what I wanted. 
```
is_success, motion_matrix, info_matrix = o3d.odometry.compute_rgbd_odometry(rgbd, new_rgbd, 
pinhole_camera_intrinsic=my_camera_intrinsics)
```
I got this warning:

```[Open3D WARNING] [RGBDOdometry] Two RGBD pairs should be same in size.```

Of course the RGBD pairs are created in exact same fashion and also RGB and D are aligned. So, I tried to see if the odometry is working by using same source and target RGBD image.

```
is_success, motion_matrix, info_matrix = o3d.odometry.compute_rgbd_odometry(rgbd, rgbd, 
pinhole_camera_intrinsic=my_camera_intrinsics)
```

I still get the warning. The `is_success` also returns `False`. What's causing this? Is this a bug?

**Environment:**
 - OS: Ubuntu 18.04
 - Python version: 3.6.8
 - Open3D version: build from 802902c9

**Comment by theNded** [comment]
Can you please check if RGB and D are with the same size? Raw frames are not with the same size, you will need to use the [decoder](https://github.com/intel-isl/Open3D/blob/master/examples/Python/ReconstructionSystem/sensors/azure_kinect_mkv_reader.py) to align them.

**Comment by scimad** [comment]
Yeah, I am trying to modify on [this](https://github.com/intel-isl/Open3D/blob/master/examples/Python/ReconstructionSystem/sensors/azure_kinect_viewer.py) example and run with `--align_depth_to_color` argument. The frames are verified to have same size (height & width).

**Comment by scimad** [comment]
@theNded, Here's the complete code for reference.

```
import argparse
import open3d as o3d
import numpy as np


class PCDViewerWithCallback:

    def __init__(self, config, device, align_depth_to_color = True):
        self.flag_exit = False
        self.align_depth_to_color = align_depth_to_color

        self.sensor = o3d.io.AzureKinectSensor(config)
        if not self.sensor.connect(device):
            raise RuntimeError('Failed to connect to sensor')

    def escape_callback(self, vis):
        self.flag_exit = True
        return False

    def run_pointcloud(self):
        glfw_key_escape = 256
        vis = o3d.visualization.VisualizerWithKeyCallback()
        vis.register_key_callback(glfw_key_escape, self.escape_callback)
        vis.create_window('viewer', 1920, 540)
        
        print("This is for points cloud visualization. Press [ESC] to exit.")

        vis_geometry_added = False
        pcd = None
        while not self.flag_exit:
            rgbd = self.sensor.capture_frame(self.align_depth_to_color)
            if rgbd is None:
                continue
                
            # my_camera_intrinsics = o3d.camera.PinholeCameraIntrinsicParameters.PrimeSenseDefault
            my_camera_intrinsics = o3d.camera.PinholeCameraIntrinsic()
            my_camera_intrinsics.set_intrinsics(1280, 720, 601.71393, 601.49011, 636.46729, 366.23587)

            # I need to do the following because of some apparent API error
            # the rgbd image formed by rgbd = self.sensor.capture_frame() above doesn't
            # seem to comply with pcd = o3d.geometry.PointCloud.create_from_rgbd_image() below
            # and gives "[CreatePointCloudFromRGBDImage] Unsupported image format"
            # so i need to recreate the rgbd image using it's own data!!

            new_rgbd = o3d.geometry.RGBDImage.create_from_color_and_depth(rgbd.color, rgbd.depth, depth_scale=1000.0, depth_trunc=3.0, convert_rgb_to_intensity=False)

            is_success, motion_matrix, info_matrix = o3d.odometry.compute_rgbd_odometry(rgbd, rgbd, pinhole_camera_intrinsic=my_camera_intrinsics)
            rgbd = new_rgbd

            if pcd != None:
                # comment the code below if only instantaneous pcd is to be visualized
                vis.remove_geometry(pcd)
                pass

            pcd = o3d.geometry.PointCloud.create_from_rgbd_image(rgbd, my_camera_intrinsics, extrinsic=motion_matrix)
            pcd.transform([[1, 0, 0, 0], [0, -1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]])

            # if not vis_geometry_added:
            # vis.add_geometry(pcd)
            vis_geometry_added = False

            if not vis_geometry_added:
                vis.add_geometry(pcd)
                vis_geometry_added = True

            vis.update_geometry()
            vis.poll_events()
            vis.update_renderer()

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Azure kinect mkv recorder.')
    parser.add_argument('--config', type=str, help='input json kinect config')
    parser.add_argument('--list',
                        action='store_true',
                        help='list available azure kinect sensors')
    parser.add_argument('--device',
                        type=int,
                        default=0,
                        help='input kinect device id')
    parser.add_argument('-a',
                        '--align_depth_to_color',
                        action='store_true',
                        help='enable align depth image to color')
    args = parser.parse_args()

    if args.list:
        o3d.io.AzureKinectSensor.list_devices()
        exit()

    if args.config is not None:
        config = o3d.io.read_azure_kinect_sensor_config(args.config)
    else:
        config = o3d.io.AzureKinectSensorConfig()

    device = args.device
    if device < 0 or device > 255:
        print('Unsupported device id, fall back to 0')
        device = 0

    v = PCDViewerWithCallback(config, 
    v.run_pointcloud()
```

**Comment by szx0112** [comment]
I got the same error. The color and depth are exact same size except rgb has 3 channels. Any updates?

**Comment by M-A-Robson** [comment]
You can get around this by creating your rgbd images with the kwarg convert_rgb_to_intensity=True

Though then you will not have color images to integrate so you may then need to recreate color versions for that step, which is annoying and inefficient.

I think that the offending code may be in Odometry.cpp
looks like it is checking for single color channel only:

`
inline bool CheckRGBDImagePair(const geometry::RGBDImage &source,
                               const geometry::RGBDImage &target) {
    return (CheckImagePair(source.color_, target.color_) &&
            CheckImagePair(source.depth_, target.depth_) &&
            CheckImagePair(source.color_, source.depth_) &&
            CheckImagePair(target.color_, target.depth_) &&
            CheckImagePair(source.color_, target.color_) &&
            source.color_.num_of_channels_ == 1 &&
            source.depth_.num_of_channels_ == 1 &&
            target.color_.num_of_channels_ == 1 &&
            target.depth_.num_of_channels_ == 1 &&
            source.color_.bytes_per_channel_ == 4 &&
            target.color_.bytes_per_channel_ == 4 &&
            source.depth_.bytes_per_channel_ == 4 &&
            target.depth_.bytes_per_channel_ == 4);
}
`
Certainly if this returns False then the error is logged.
I tried changing the c++ file but with little luck on getting it to work with color rgbd images. Maybe a more experienced c++ programmer will be able to look at it.

**Comment by theNded** [comment]
Fixed in #1690.
